FreeRTOS/Test/CMock/queue/generic/queue_receive_nonblocking_utest.c

```diff
@@ -0,0 +1,869 @@
+
+    /* peek from the queue while empty */
+    TEST_ASSERT_EQUAL( pdFALSE, xQueuePeek( xQueue, &checkVal, 0 ) );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ * @brief Test xQueuePeek with a full queue of length 1, items size 0
+ * @coverage xQueuePeek
+ */
+void test_xQueuePeek_zeroItemSize_full( void )
+{
+    /* Create a new queue */
+    QueueHandle_t xQueue = xQueueCreate( 1, 0 );
+
+    xQueueSend( xQueue, NULL, 0 );
+
+    /* peek from the queue while full */
+    TEST_ASSERT_EQUAL( pdTRUE, xQueuePeek( xQueue, NULL, 0 ) );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ * @brief Test xQueuePeek with a null pvBuffer on an empty queue of length 1, items size 4
+ * @coverage xQueuePeek
+ */
+void test_xQueuePeek_fourItemSize_nullptr_assert_empty( void )
+{
+    /* Create a new queue */
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    fakeAssertExpectFail();
+
+    /* peek from the queue with a nullptr storage location */
+    TEST_ASSERT_EQUAL( pdFALSE, xQueuePeek( xQueue, NULL, 0 ) );
+
+    TEST_ASSERT_EQUAL( true, fakeAssertGetFlagAndClear() );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ * @brief Test xQueuePeek with a null pvBuffer on a full queue of length 1, items size 4
+ * @coverage xQueuePeek
+ */
+void test_xQueuePeek_fourItemSize_nullptr_assert_full( void )
+{
+    /* Create a new queue */
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    uint32_t testVal = getNextMonotonicTestValue();
+
+    xQueueSend( xQueue, &testVal, 0 );
+
+    /* peek from the queue with a nullptr storage location */
+    EXPECT_ASSERT_BREAK( xQueuePeek( xQueue, NULL, 0 ) );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ * @brief Test xQueuePeek with an occupied queue
+ * @coverage xQueuePeek
+ */
+void test_xQueuePeek_success( void )
+{
+    /* Create a new queue */
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    uint32_t testVal = getNextMonotonicTestValue();
+
+    xQueueSend( xQueue, &testVal, 0 );
+
+    uint32_t checkVal = INVALID_UINT32;
+
+    /* peek from the queue while full */
+    TEST_ASSERT_EQUAL( pdTRUE, xQueuePeek( xQueue, &checkVal, 0 ) );
+    TEST_ASSERT_EQUAL( testVal, checkVal );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ * @brief Test xQueuePeek with an occupied queue with higher priority tasks waiting.
+ * @coverage xQueuePeek
+ */
+void test_xQueuePeek_tasks_waiting_higher_priority_success( void )
+{
+    /* Create a new queue */
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    /* Add a value to the queue */
+    uint32_t testVal = getNextMonotonicTestValue();
+
+    xQueueSend( xQueue, &testVal, 0 );
+
+    /* Insert an item into the event list */
+    td_task_setFakeTaskPriority( DEFAULT_PRIORITY + 1 );
+    td_task_addFakeTaskWaitingToReceiveFromQueue( xQueue );
+
+    /* peek from the queue */
+    uint32_t checkVal = INVALID_UINT32;
+
+    TEST_ASSERT_EQUAL( pdTRUE, xQueuePeek( xQueue, &checkVal, 0 ) );
```

Comment:
Does xQueuePeek fail in this case ?

Reply:
Good point. I'll add some test cases to cover this a bit better.

Reply:
After reviewing this a bit more, xQueuePeek does not fail in this case because it does not Yield until after performing the Peek operation. xQueueReceive works in a similar way.